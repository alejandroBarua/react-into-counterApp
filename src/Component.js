import React from 'react';
import PropTypes from 'prop-types';

const Component = ({ title = 'title', subtitle }) => {

	return (

		<> 
			<h1> { title } </h1>
			<p> { subtitle } </p>
		</>

	);
} 

Component.propTypes = {
	title: PropTypes.string.isRequired 
}

Component.defaultProps = {
	subtitle: 'This is My first React App'
}

export default Component; 