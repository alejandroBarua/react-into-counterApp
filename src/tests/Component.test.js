import React from 'react';
//import { render } from '@testing-library/react';
import Component from '../Component';
import { shallow } from 'enzyme';

describe('tests <Component />', () => {
	
/* 	test('render title', () => {
		
		const title = "Hello World";
		const { getByText } = render(<Component title={title} />);

		expect(getByText(title)).toBeInTheDocument();
	}) */

	test('render title', () => {
		
		const title = "Hello World";
		const wrapper = shallow(<Component title={title} />);

		expect(wrapper).toMatchSnapshot();

	})

	test('render subtitle', () => {
		
		const title = "Hello World";
		const subtitle = "I am a subtitle";
		const wrapper = shallow(
			<Component 
				title={title} 
				subtitle={subtitle}
			/>);

		const textParagraph = wrapper.find('p').text(); // funciona como el querySelector('#id')

		expect(textParagraph).toBe(subtitle);

	})
})
