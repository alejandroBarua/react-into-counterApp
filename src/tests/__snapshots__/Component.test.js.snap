// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests <Component /> render title 1`] = `
<Fragment>
  <h1>
    Hello World
  </h1>
  <p>
    This is My first React App
  </p>
</Fragment>
`;
